### PR TITLE
[PR] ForwardsDamageMultiplier + tweak for #1592

### PR DESCRIPTION
**Pull request by boot2big**
ForwardsDamageMultiplier will, as the name implies, multiply damage forwarded by whatever it is defined as. 0.5 will deal half damage to the vehicle, 2 will do twice as much and 5 is right out. 
If not defined, LCs will set this to either 1 or 4, depending on whether the part isn't or is an engine. (The logic being that in (newer versions of) GTA, shooting the engine of a vehicle damages it much more than shooting the doors or body panels.)

### CODE REVIEW

File: mccore/src/main/java/minecrafttransportsimulator/jsondefs/JSONPart.java [hunk 1]
```diff
@@ -99,6 +99,9 @@ public static class JSONPartGeneric {
         @JSONDescription("If true, this part will forward damage onto the vehicle it is on when hit by a bullet.  This will also cause the bullet to stop when it hits this part.  Engines ignore this behavior and always forward damage.")
         public boolean forwardsDamage;
 
+        @JSONDescription("The amount to multiply forwarded damage by, for more or less sensitive parts. Defaults to a whopping 1x multiplier if not defined.")
```
Comment (DonBruce64) [inline mccore/src/main/java/minecrafttransportsimulator/jsondefs/JSONPart.java:102]: Descriptions in technical documentation should be boring and simple to not distract.

File: mccore/src/main/java/minecrafttransportsimulator/packloading/LegacyCompatSystem.java [hunk 2]
```diff
@@ -555,7 +555,12 @@ private static void performPartLegacyCompats(JSONPart definition) {
             definition.generic.useVehicleTexture = definition.general.useVehicleTexture;
             definition.general.useVehicleTexture = false;
         }
-
+        
+        //While we're here in generic, check if we have forwardsDamageMultiplier defined and if not, lemon-grenade it until it's defined.
+        if (definition.generic.forwardsDamageMultiplier == 0) {
+                definition.generic.forwardsDamageMultiplier = definition.generic.type.contains("engine") ? 4 : 1;//If this part is an engine, deal more damage to the vehicle by default (unless that FDM was defined, of course)
```
Comment (DonBruce64) [inline mccore/src/main/java/minecrafttransportsimulator/packloading/LegacyCompatSystem.java:561]: I'm un-sure about this.  This is something @Ilya-torshov and @LauraDarkez should probably weigh in on.

**Comment by LauraDarkez** [comment]
what am i being asked about?

**Comment by LauraDarkez** [comment]
okay, i dont fully understand the purpose of this. is this supposed to be a default or a opt-in via json change?

**Comment by LauraDarkez** [comment]
okay, i dont understand why the engine taking damage would DEFAULTLY deal more damage to the vehicle. this seems like a bad idea

**Comment by boot2big** [comment]
It was inspiration from GTA IV/V, where shooting the front of a vehicle (or for rear-engine vehicles, the rear) will deal a lot more damage to the vehicle as that is where important bits like fuel lines, the intercooler and the engine tend to be.

**Comment by LauraDarkez** [comment]
okay, well to be honest i think it should NOT default to be higher, and instead should just be an optional json line you can put in

**Comment by DonBruce64** [comment]
Works for me.  Just make sure to add the paramter to the wiki.

**Review by DonBruce64 — COMMENTED** [review]
Indentation is still as wonky as a bus driving in a forest, but it'll do well enough I suppose.  Just put the wiki info in and I'll pull this change in.

**Review by DonBruce64 — APPROVED** [review]

**Comment by boot2big** [comment]
https://github.com/DonBruce64/MinecraftTransportSimulator/wiki/Pack-Making-JSON-Parts
I liked the part where he said "it's wiki'ing time" and he wiki'd all over this page!

**Pull request merged**